Use a C stub to call the uname function from the C standard library instead of calling the uname POSIX command (#6217)
ocaml/opam: master_changes.md

## opam-format

> ## opam-core

Reviewer: Maybe it worth having an entry in `#Internal`, to highlight the better system polling change

Author: done

>   * `OpamStd.Sys.{get_terminal_columns,guess_shell_compat}`: Harden the process calls to account for failures [#6230 @kit-ty-kate - fix #6215]

Reviewer: ```suggestion
```

>   * `OpamStd.Sys.{get_terminal_columns,guess_shell_compat}`: Harden the process calls to account for failures [#6230 @kit-ty-kate - fix #6215]

Reviewer: ```suggestion
  * `OpamStd.Sys.{get_terminal_columns,uname,getconf,guess_shell_compat}`: Harden the process calls to account for failures [#6230 @kit-ty-kate - fix #6215]
```
No need to update it, uname & getconf are part of that PR

  * `OpamStd.Sys.getconf`: was removed [#6217 @kit-ty-kate]

>   * `OpamStd.Sys.uname`: now returns the memoized result of `uname(2)` [#6217 @kit-ty-kate]

Reviewer: In some distros, the `uname` C function is `uname(3)`. I think its better to refer it in plain text.
```suggestion
  * `OpamStd.Sys.uname`: now returns the memoized result of `uname` C binding [#6217 @kit-ty-kate]
```

>   * `OpamStd.Sys.getconf`: was removed [#6217 @kit-ty-kate]
>   * `OpamStd.Sys.uname`: now returns the memoized result of `uname(2)` [#6217 @kit-ty-kate]
>   * `OpamStd.Sys.uname_freebsd_version`: was added, which returns the output of the `uname -U` command [#6217 @kit-ty-kate]
>   * `OpamStd.Sys.getconf_long_bit`: was added, which returns the output of the `getconf LONG_BIT` command [#6217 @kit-ty-kate]

Reviewer: ```suggestion
  * `OpamStd.Sys.getconf`: was removed, replaced by `get_conf_long_bit` [#6217 @kit-ty-kate]
  * `OpamStd.Sys.getconf_long_bit`: was added, which returns the output of the `getconf LONG_BIT` command [#6217 @kit-ty-kate]
  * `OpamStd.Sys.uname`: now returns the memoized result of `uname(2)` [#6217 @kit-ty-kate]
  * `OpamStd.Sys.uname_freebsd_version`: was added, which returns the output of the `uname -U` command [#6217 @kit-ty-kate]
```
